### ISSUE TITLE
How to shut down a countdown instance?

### ISSUE THREAD

**Issue by felixchan** [kbwood/countdown#48]
I'd like to properly shutdown the instance before removing the component from the dom.

Is this possible?

Right now, I'm using Backbone, and if I do not shut it down, I get an error because the timer keeps running.

```
Uncaught TypeError: Cannot read property 'significant' of undefined in jquery.countdown.js:573
```

**Comment by tapaspalui** [comment]
Getting same on load the page once again on ajax

**Comment by kbwood** [comment]
Can you provide a page that shows the problem?

**Comment by deanpapastrat** [comment]
I'm having a similar problem - I'm using Ruby on Rails with Turbolinks, and when I navigate to other pages using Turbolinks, the countdown throws the error "cannot read property significant of undefined". I believe this is because the countdown isn't releasing when I call the AJAX page load via Turbolinks.

**Comment by kbwood** [comment]
Can you provide a page that shows the problem?

**Comment by deanpapastrat** [comment]
I only have it running locally. I can send my Coffeescript file that initializes the countdown if that's of any use.

**Comment by IlyaDonskikh** [comment]
When i reload my page with Ajax i have same error (Ruby on Rails without turbolinks):

```
Uncaught TypeError: Cannot read property 'significant' of undefined
```

Code:
```js
$('.pincode_countdown_JS').countdown
  until: countdownTime
  compact: true
  format: 'MS'
  description: ''
  onExpiry: tryAgainPincode
```

**Comment by isuvorov** [comment]
+1

**Comment by isuvorov** [comment]
there is example:
http://coder24.ru/countdown/

Error in console

**Comment by kbwood** [comment]
This example doesn't throw the error listed above. Also, the countdown can only be initialised once, so the subsequent calls in _reload are ignored.

**Comment by flyingQuQu** [comment]
I hava the same problem above while using ajax to refill the dom which contains the countdown element.

**Comment by kbwood** [comment]
Can you provide a page that shows the problem?

**Comment by kynitssa** [comment]
the same problem. it`s on 606 line: Uncaught TypeError: Cannot read property 'significant' of undefined. console.log gives me 0, 0 again and 0 goes counting then seconds starts countdown, but if I'm chang to another date without page reload, it gives me 0, 0 again and then error. inst.options.significant goes undefined. I can't provide u page with problem, hope u repare this soon

**Comment by brovikov** [comment]
+1 Uncaught TypeError: Cannot read property 'significant' of undefined, on 606 line, when ajax partial reload (Ruby on Rails 4.2 without turbolinks)

**Comment by kbwood** [comment]
Can you provide a page that shows the problem?

**Comment by dmitry-zaporozhan** [comment]
Same when calling `$(slector).countdown(config)` for second time (after partial page update to re-init countdown). Using version 2.0.1. Works on 2.0.0 though.

**Comment by umeshm11** [comment]
I have also occurs this problem after ajax call getting error :- Uncaught TypeError: Cannot read property 'significant' of undefined

I solve this error using -
before ajax call i have used  $('.cntDown').countdown('toggle');  

cntDown is multiple instance with same class.

**Comment by krpraveen** [comment]
This worked for me.
$(selector).countdown('destroy') ;

**Comment by satyeshu** [comment]
I was also facing the same thing (Cannot read the property 'significant' of undefined). "$(selector).countdown('destroy');" worked for me too.

**Comment by sagarrakshe** [comment]
I am using the plugin in angular directive as follows, and when I re-visit the same view, I get `Cannot read property 'significant' of undefined`. 

```
UserApplication.directive('jqCountdown', [function () {
    return {
      restrict: 'A',
      scope: {
        'model': '=',
        'finish': '&'
      },
      link: function(scope, elem, attrs) {
        $(elem).countdown({
          until: +scope.model,
          layout:'<b>{d<}{dn}d {d>} {hnn}:{mnn}:{snn}</b>',
          onExpiry: function () {
            scope.finish();
          }
        });
      }
    };
  }]);
```

**Comment by krpraveen** [comment]
sagarrakshe ! have you tried using `$(selector).countdown('destroy') ;` before disposing the view. Actually All instances of the count down timer MUST be disposed correctly before the page is disposed.  And the correct way to dispose is `$(selector).countdown('destroy') ;` the Otherwise, this problem occurs.

**Comment by sagarrakshe** [comment]
Yes, `$(selector).countdown('destroy')` did work, at some point my timer value was getting `undefined` and hence it was giving me error. Thanks!

**Comment by Jeremlicious** [comment]
I can't where I've missed a `$(selector).countdown('destroy')`
I think this plugin should help finding what coundown is missing by throwing a meaningful error
I made an attempt here : https://github.com/Jeremlicious/countdown/commit/6cf39e3300fc380f326571ea9c82b7747d7f51d1
I'll let you know if it helps.

**Comment by mmmveggies** [comment]
I got `Uncaught TypeError: Cannot read property 'significant' of undefined` a lot. The only way I was able to solve it was going from 
```js
$(`<div />`).countdown({ until: +300 });
```
to
```js
setTimeout(() => {
  $(`<div />`).countdown({ until: +300 })
}, 1);
```

Due to the fact I was trying to
